packages/dd-trace/src/scope/base.js

```diff
@@ -140,11 +143,14 @@ function wrapThen (then, scope, span) {
   }
 }
 
-function wrapAddListener (addListener, scope, span) {
-  return function addListenerWithTrace (eventName, listener) {
-    if (!listener || listener._datadog_unbound || listener.listener) {
+function wrapAddListener (fnName, obj) {
```

Comment:
Why change that this usually returns the function? It's a common pattern for wrappers in the tracer, inspired by `shimmer`.

Reply:
Honestly, that approach just seemed more complicated to me.

Reply:
That's fine then, I just wanted to make sure code is not changed unnecessarily.

Reply:
That being said, `wrapMethod` is there so I might as well use it. Fixed now.